src/main/java/com/nike/cerberus/client/auth/aws/InstanceRoleVaultCredentialsProvider.java

```diff
@@ -152,8 +152,16 @@ protected String getInstanceProfileArn() {
         final String accountId = instanceProfileInfo.accountId;
         final String path = parsePathFromInstanceProfileName(instanceProfileInfo.profileName);
 
+        // There isn't a 100% reliable method for constructing the role ARN from the meta-data endpoint.
+        // So here we try both with and without the path that was used in the instanceProfileArn.
+        // The only reason we don't try and auth with the instanceProfileArn is it isn't a valid ARN type
+        // that can be included in a KMS key policy.
         for (String roleName : securityCredentialsKeySet) {
             result.add(buildRoleArn(accountId, path, roleName));
+            if (path != null) {
```

Comment:
I keep getting confused by this code. Why is this a ___not___ equals check again, rather than ___equals___?

Reply:
I think it's because: if path is null then it has already been sent in the first call

Reply:
nit: I might find it less confusing if it was:
```
for (String roleName : securityCredentialsKeySet) {
    result.add(buildRoleArn(accountId, null, roleName));
    if (path != null) {
        result.add(buildRoleArn(accountId, path, roleName));
    }
```

Reply:
The for loop is creating roleArns both with and without the path found in the instanceProfileArn.  This is kind of a messy way to get what we want.  The roleArn seems to be completely independent of the instanceProfileArn but if you have a path in one you might be using it in the other (just by convention).  I think there is technically another edge case where the roleArn and instanceProfileArn could actually use different paths.

If path was null then we already called buildRoleArn() with a null path on the line before so we don't need to do it a second time.

Reply:
I get the nitpick, it is a little better.  I made the change but altered the order so that we'll try with paths (seems to be more common) before the without path case.